content/blog/2024-06-15-rustnl-2024-unconference.md

```diff
@@ -0,0 +1,203 @@
+
+Rik asked if there was a minimal set of ARIA attributes that people could start using in their UI, as the set of available attributes is gigantic.
+The AccessKit framework is rich, and it's not obvious at first glance how much is needed versus nice-to-have.
+The current reference in the Rust ecosystem is [the AccessKit implementation in eGUI](https://github.com/emilk/egui/pull/2294).
+
+### Requests for Rust features
+
+Jon Kelley, the creator of Dioxus, had [a laundry list of features](https://dioxus.notion.site/Dioxus-Labs-High-level-Rust-5fe1f1c9c8334815ad488410d948f05e) he wanted from the Rust language.
+That list was later [filed in the Project Goals repository]((https://github.com/rust-lang/rust-project-goals/pull/10)).
+
+The requests were well-documented, with motivating examples in existing Dioxus code that would clearly be improved by the features.
+Most of them felt both self-contained and like they would tremendously both beginners and veteran Rust developers; though Jon made the point that the papercuts he wanted to remove were especially punishing for new developers trying to become familiar with Rust.
+
+Overall people were very enthusiastic about them, and the lang team members in the room seemed pretty optimistic that those features could be added.
+
+My favorite proposals were:
+
+- `Capture` trait for auto-cloning reference-counted types in closures.
+- Automatic partial borrows.
+- Optional struct attributes.
+
+Jon also wanted support for a remote cache of pre-built crates, so that users wouldn't suffer from having to rebuild dozens of crates every time you download your crate's dependencies.
+The consensus from Rust maintainers was that such a cache would be essentially impossible with Rust's current architecture.
+
+Niko Matsakis recently released [The borrow checker within](https://smallcultfollowing.com/babysteps/blog/2024/06/02/the-borrow-checker-within/), which feels like a follow-up to some of these discussions.
+
+
+## Collaboration in the Rust GUI ecosystem
+
+Over the last two years I've joked a few times that everybody in the Rust GUI ecosystem wants the ecosystem to standardize on common crates, as long as it's *their* crates.
+
+After RustNL, I'm pleased to report this joke isn't accurate anymore.
+
+A vibe that permeated all discussions was people wanting work to be pooled.
+It felt like people didn't strongly care whose crate we adopted, as long as we all agreed the crate made sense:
+we're slowly crawling out of the [xkcd 927](https://xkcd.com/927/) dilemma.
+
+The main ecosystem crates that people are enthusiastic about sharing are:
+
+- [winit](https://github.com/rust-windowing/winit) for creating windows.
+- [AccessKit](https://github.com/AccessKit/accesskit/) for plugging into accessibility APIs.
+- [Parley](https://github.com/linebender/parley) for rendering text and as a basis for a text-editing widget.
```

Comment:
In discussions since the unconference, Bevy has (correctly) decided to go with cosmic-text for text layout.

If you want a third example, I think wgpu is *probably* a better candidate.